# First Half Data of the 2018-2019 Season in the English Premier League

In [1]:
import pandas as pd
import numpy as np
import warnings
warnings.simplefilter(action='ignore', category=FutureWarning)

In [2]:
df = pd.read_csv("season-1819.csv")
df.head()

Date      HomeTeam        AwayTeam  FTHG  FTAG FTR  HTHG  HTAG HTR  \
0  10/08/18    Man United       Leicester     2     1   H     1     0   H   
1  11/08/18   Bournemouth         Cardiff     2     0   H     1     0   H   
2  11/08/18        Fulham  Crystal Palace     0     2   A     0     1   A   
3  11/08/18  Huddersfield         Chelsea     0     3   A     0     2   A   
4  11/08/18     Newcastle       Tottenham     1     2   A     1     2   A   

      Referee  ...  HST  AST  HF  AF  HC  AC  HY  AY  HR  AR  
0  A Marriner  ...    6    4  11   8   2   5   2   1   0   0  
1    K Friend  ...    4    1  11   9   7   4   1   1   0   0  
2      M Dean  ...    6    9   9  11   5   5   1   2   0   0  
3  C Kavanagh  ...    1    4   9   8   2   5   2   1   0   0  
4  M Atkinson  ...    2    5  11  12   3   5   2   2   0   0  

[5 rows x 22 columns]

In [3]:
df.info()

<class 'pandas.core.frame.DataFrame'>
RangeIndex: 210 entries, 0 to 209
Data columns (total 22 columns):
 #   Column    Non-Null Count  Dtype 
---  ------    --------------  ----- 
 0   Date      210 non-null    object
 1   HomeTeam  210 non-null    object
 2   AwayTeam  210 non-null    object
 3   FTHG      210 non-null    int64 
 4   FTAG      210 non-null    int64 
 5   FTR       210 non-null    object
 6   HTHG      210 non-null    int64 
 7   HTAG      210 non-null    int64 
 8   HTR       210 non-null    object
 9   Referee   210 non-null    object
 10  HS        210 non-null    int64 
 11  AS        210 non-null    int64 
 12  HST       210 non-null    int64 
 13  AST       210 non-null    int64 
 14  HF        210 non-null    int64 
 15  AF        210 non-null    int64 
 16  HC        210 non-null    int64 
 17  AC        210 non-null    int64 
 18  HY        210 non-null    int64 
 19  AY        210 non-null    int64 
 20  HR        210 non-null    int64 
 21  AR        210 no

## Meaning of the abbreviations

- FTHG = Full Time Home Team Goals
- FTAG = Full Time Away Team Goals
- FTR = Full Time Result (shows the winning team) ("H" stands for "Home", "D" stands for "Draw", "A" stands for "Away")
- HTHG = Half Time Home Team Goals
- HTAG = Half Time Away Team Goals
- HTR = Half Time Result (similar to "FTR")
- HS = Home Team Shots
- AS = Away Team Shots
- HST = Home Team Shots on Target
- AST = Away Team Shots on Target
- HF = Home Team Fouls Committed
- AF = Away Team Fouls Committed
- HC = Home Team Corners
- AC = Away Team Corners
- HY = Home Team Yellow Cards
- AY = Away Team Yellow Cards
- HR = Home Team Red Cards
- AR = Away Team Red Cards

## Getting only the first half data
    When we call the info() method it says there are 210 entries in the dataframe. That means there are 210 rows, so this dataframe includes 210 matches which played between the dates 10/08/18 and 03/01/19. There are 20 teams in the English Premier League and that means there should 10 matches be played each week. Every Team plays 19 matches in the first half of the season. In conclusion this dataframe should have 190 entries to include only the matches which played in the first half of the season.
    
    For the correct implementation of this idea we will create an another column named "Match", which will combine the teams playing the match into a list. The reason we have not simply dropped the last 20 rows is to check if there are any matches be postponed by the FA due to the scheduling reasons.

In [4]:
df["Match"] = df.apply(lambda row: tuple(sorted([row["HomeTeam"], row["AwayTeam"]])), axis=1) 
df

Date      HomeTeam        AwayTeam  FTHG  FTAG FTR  HTHG  HTAG HTR  \
0    10/08/18    Man United       Leicester     2     1   H     1     0   H   
1    11/08/18   Bournemouth         Cardiff     2     0   H     1     0   H   
2    11/08/18        Fulham  Crystal Palace     0     2   A     0     1   A   
3    11/08/18  Huddersfield         Chelsea     0     3   A     0     2   A   
4    11/08/18     Newcastle       Tottenham     1     2   A     1     2   A   
..        ...           ...             ...   ...   ...  ..   ...   ...  ..   
205  02/01/19  Huddersfield         Burnley     1     2   A     1     1   D   
206  02/01/19     Newcastle      Man United     0     2   A     0     0   D   
207  02/01/19      West Ham        Brighton     2     2   D     0     0   D   
208  02/01/19        Wolves  Crystal Palace     0     2   A     0     0   D   
209  03/01/19      Man City       Liverpool     2     1   H     1     0   H   

        Referee  ...  AST  HF  AF  HC  AC  HY  AY  HR  AR  \
0    A Marriner  ...    4  11   8   2   5   2   1   0   0   
1      K Friend  ...    1  11   9   7   4   1   1   0   0   
2        M Dean  ...    9   9  11   5   5   1   2   0   0   
3    C Kavanagh  ...    4   9   8   2   5   2   1   0   0   
4    M Atkinson  ...    5  11  12   3   5   2   2   0   0   
..          ...  ...  ...  ..  ..  ..  ..  ..  ..  ..  ..   
205      M Dean  ...    8  11   9   5   2   0   4   1   1   
206  A Marriner  ...    7  10   9   1   2   1   2   0   0   
207  C Kavanagh  ...    6   9  11   1   5   0   1   0   0   
208      R East  ...    4   9   7   3  10   4   1   0   0   
209    A Taylor  ...    5  12   7   2   1   4   2   0   0   

                        Match  
0     (Leicester, Man United)  
1      (Bournemouth, Cardiff)  
2    (Crystal Palace, Fulham)  
3     (Chelsea, Huddersfield)  
4      (Newcastle, Tottenham)  
..                        ...  
205   (Burnley, Huddersfield)  
206   (Man United, Newcastle)  
207      (Brighton, West Ham)  
208  (Crystal Palace, Wolves)  
209     (Liverpool, Man City)  

[210 rows x 23 columns]

In [5]:
df = df.drop_duplicates(subset="Match", keep="first")
df

Date      HomeTeam        AwayTeam  FTHG  FTAG FTR  HTHG  HTAG HTR  \
0    10/08/18    Man United       Leicester     2     1   H     1     0   H   
1    11/08/18   Bournemouth         Cardiff     2     0   H     1     0   H   
2    11/08/18        Fulham  Crystal Palace     0     2   A     0     1   A   
3    11/08/18  Huddersfield         Chelsea     0     3   A     0     2   A   
4    11/08/18     Newcastle       Tottenham     1     2   A     1     2   A   
..        ...           ...             ...   ...   ...  ..   ...   ...  ..   
185  26/12/18     Liverpool       Newcastle     4     0   H     1     0   H   
186  26/12/18    Man United    Huddersfield     3     1   H     1     0   H   
187  26/12/18     Tottenham     Bournemouth     5     0   H     3     0   H   
188  26/12/18       Watford         Chelsea     1     2   A     1     1   D   
189  27/12/18   Southampton        West Ham     1     2   A     0     0   D   

        Referee  ...  AST  HF  AF  HC  AC  HY  AY  HR  AR  \
0    A Marriner  ...    4  11   8   2   5   2   1   0   0   
1      K Friend  ...    1  11   9   7   4   1   1   0   0   
2        M Dean  ...    9   9  11   5   5   1   2   0   0   
3    C Kavanagh  ...    4   9   8   2   5   2   1   0   0   
4    M Atkinson  ...    5  11  12   3   5   2   2   0   0   
..          ...  ...  ...  ..  ..  ..  ..  ..  ..  ..  ..   
185     G Scott  ...    2   7   9  10   2   0   0   0   0   
186      J Moss  ...    2   9  13   5   3   1   1   0   0   
187  C Kavanagh  ...    4   4   8   3   4   2   1   0   0   
188  M Atkinson  ...    4  15   5   3   4   1   0   0   0   
189    C Pawson  ...    5  10   3   4   5   2   0   0   0   

                          Match  
0       (Leicester, Man United)  
1        (Bournemouth, Cardiff)  
2      (Crystal Palace, Fulham)  
3       (Chelsea, Huddersfield)  
4        (Newcastle, Tottenham)  
..                          ...  
185      (Liverpool, Newcastle)  
186  (Huddersfield, Man United)  
187    (Bournemouth, Tottenham)  
188          (Chelsea, Watford)  
189     (Southampton, West Ham)  

[190 rows x 23 columns]

In [6]:
df = df.drop(columns="Match")
df

Date      HomeTeam        AwayTeam  FTHG  FTAG FTR  HTHG  HTAG HTR  \
0    10/08/18    Man United       Leicester     2     1   H     1     0   H   
1    11/08/18   Bournemouth         Cardiff     2     0   H     1     0   H   
2    11/08/18        Fulham  Crystal Palace     0     2   A     0     1   A   
3    11/08/18  Huddersfield         Chelsea     0     3   A     0     2   A   
4    11/08/18     Newcastle       Tottenham     1     2   A     1     2   A   
..        ...           ...             ...   ...   ...  ..   ...   ...  ..   
185  26/12/18     Liverpool       Newcastle     4     0   H     1     0   H   
186  26/12/18    Man United    Huddersfield     3     1   H     1     0   H   
187  26/12/18     Tottenham     Bournemouth     5     0   H     3     0   H   
188  26/12/18       Watford         Chelsea     1     2   A     1     1   D   
189  27/12/18   Southampton        West Ham     1     2   A     0     0   D   

        Referee  ...  HST  AST  HF  AF  HC  AC  HY  AY  HR  AR  
0    A Marriner  ...    6    4  11   8   2   5   2   1   0   0  
1      K Friend  ...    4    1  11   9   7   4   1   1   0   0  
2        M Dean  ...    6    9   9  11   5   5   1   2   0   0  
3    C Kavanagh  ...    1    4   9   8   2   5   2   1   0   0  
4    M Atkinson  ...    2    5  11  12   3   5   2   2   0   0  
..          ...  ...  ...  ...  ..  ..  ..  ..  ..  ..  ..  ..  
185     G Scott  ...    8    2   7   9  10   2   0   0   0   0  
186      J Moss  ...   10    2   9  13   5   3   1   1   0   0  
187  C Kavanagh  ...    7    4   4   8   3   4   2   1   0   0  
188  M Atkinson  ...    2    4  15   5   3   4   1   0   0   0  
189    C Pawson  ...    5    5  10   3   4   5   2   0   0   0  

[190 rows x 22 columns]

## Creating the first half standings of the season with the data we have

- GF = Goals For
- GA = Goals Against
- GD = Goal Difference

In [7]:
teams = ["Man United", "Man City", "Liverpool", "Chelsea", "Bournemouth",\
         "Watford", "Southampton", "Burnley", "Huddersfield", "Newcastle",\
         "West Ham", "Brighton", "Crystal Palace", "Fulham", "Arsenal",\
        "Wolves", "Tottenham", "Cardiff", "Everton", "Leicester"]
position = ['1','2','3','4','5','6','7','8','9','10','11','12','13','14','15','16','17','18','19','20']
table_df = pd.DataFrame(sorted(teams), columns=["Club"], index=position)
table_df[["Won","Drawn","Lost","GF","GA","GD","Points"]] = 0
table_df

Club  Won  Drawn  Lost  GF  GA  GD  Points
1          Arsenal    0      0     0   0   0   0       0
2      Bournemouth    0      0     0   0   0   0       0
3         Brighton    0      0     0   0   0   0       0
4          Burnley    0      0     0   0   0   0       0
5          Cardiff    0      0     0   0   0   0       0
6          Chelsea    0      0     0   0   0   0       0
7   Crystal Palace    0      0     0   0   0   0       0
8          Everton    0      0     0   0   0   0       0
9           Fulham    0      0     0   0   0   0       0
10    Huddersfield    0      0     0   0   0   0       0
11       Leicester    0      0     0   0   0   0       0
12       Liverpool    0      0     0   0   0   0       0
13        Man City    0      0     0   0   0   0       0
14      Man United    0      0     0   0   0   0       0
15       Newcastle    0      0     0   0   0   0       0
16     Southampton    0      0     0   0   0   0       0
17       Tottenham    0      0     0   0   0   0       0
18         Watford    0      0     0   0   0   0       0
19        West Ham    0      0     0   0   0   0       0
20          Wolves    0      0     0   0   0   0       0

In [8]:
# Creating three different dataframes by the full time results of the matches
home_winner = df[df["FTR"] == "H"] 
away_winner = df[df["FTR"] == "A"]
draw = df[df["FTR"] == "D"]

# Increasing the number in columns based on the data frames created above
for team in home_winner["HomeTeam"]:
    table_df.loc[table_df["Club"] == team, "Won"] += 1
for team in home_winner["AwayTeam"]:
    table_df.loc[table_df["Club"] == team, "Lost"] += 1
    
for team in away_winner["AwayTeam"]:
    table_df.loc[table_df["Club"] == team, "Won"] += 1
for team in away_winner["HomeTeam"]:
    table_df.loc[table_df["Club"] == team, "Lost"] += 1
    
for _, row in draw.iterrows():
    for team in [row["HomeTeam"], row["AwayTeam"]]:
        table_df.loc[table_df["Club"] == team, "Drawn"] += 1

table_df
    
    
            
            
        

Club  Won  Drawn  Lost  GF  GA  GD  Points
1          Arsenal   11      5     3   0   0   0       0
2      Bournemouth    8      2     9   0   0   0       0
3         Brighton    6      4     9   0   0   0       0
4          Burnley    3      3    13   0   0   0       0
5          Cardiff    4      3    12   0   0   0       0
6          Chelsea   12      4     3   0   0   0       0
7   Crystal Palace    5      4    10   0   0   0       0
8          Everton    7      6     6   0   0   0       0
9           Fulham    2      5    12   0   0   0       0
10    Huddersfield    2      4    13   0   0   0       0
11       Leicester    8      4     7   0   0   0       0
12       Liverpool   16      3     0   0   0   0       0
13        Man City   14      2     3   0   0   0       0
14      Man United    9      5     5   0   0   0       0
15       Newcastle    4      5    10   0   0   0       0
16     Southampton    3      6    10   0   0   0       0
17       Tottenham   15      0     4   0   0   0       0
18         Watford    8      3     8   0   0   0       0
19        West Ham    8      3     8   0   0   0       0
20          Wolves    7      5     7   0   0   0       0

In [9]:
# Calculating the points of the teams and sorting by the points
table_df["Points"] = table_df["Won"] * 3 + table_df["Drawn"]
table_df = table_df.sort_values(by="Points", ascending = False)

table_df.index = position
table_df
        
        

Club  Won  Drawn  Lost  GF  GA  GD  Points
1        Liverpool   16      3     0   0   0   0      51
2        Tottenham   15      0     4   0   0   0      45
3         Man City   14      2     3   0   0   0      44
4          Chelsea   12      4     3   0   0   0      40
5          Arsenal   11      5     3   0   0   0      38
6       Man United    9      5     5   0   0   0      32
7        Leicester    8      4     7   0   0   0      28
8          Everton    7      6     6   0   0   0      27
9          Watford    8      3     8   0   0   0      27
10        West Ham    8      3     8   0   0   0      27
11     Bournemouth    8      2     9   0   0   0      26
12          Wolves    7      5     7   0   0   0      26
13        Brighton    6      4     9   0   0   0      22
14  Crystal Palace    5      4    10   0   0   0      19
15       Newcastle    4      5    10   0   0   0      17
16         Cardiff    4      3    12   0   0   0      15
17     Southampton    3      6    10   0   0   0      15
18         Burnley    3      3    13   0   0   0      12
19          Fulham    2      5    12   0   0   0      11
20    Huddersfield    2      4    13   0   0   0      10

In [10]:
# Creating a dataframe to calculate the Goal Difference
goals = df[["HomeTeam", "AwayTeam", "FTHG", "FTAG"]]

goals

HomeTeam        AwayTeam  FTHG  FTAG
0      Man United       Leicester     2     1
1     Bournemouth         Cardiff     2     0
2          Fulham  Crystal Palace     0     2
3    Huddersfield         Chelsea     0     3
4       Newcastle       Tottenham     1     2
..            ...             ...   ...   ...
185     Liverpool       Newcastle     4     0
186    Man United    Huddersfield     3     1
187     Tottenham     Bournemouth     5     0
188       Watford         Chelsea     1     2
189   Southampton        West Ham     1     2

[190 rows x 4 columns]

In [11]:
for _,row in goals.iterrows():
    home_team = row["HomeTeam"]
    away_team = row["AwayTeam"]
    home_goals = row["FTHG"]
    away_goals = row["FTAG"]

    table_df.loc[table_df["Club"] == home_team, "GF"] += home_goals
    table_df.loc[table_df["Club"] == home_team, "GA"] += away_goals
    table_df.loc[table_df["Club"] == away_team, "GF"] += away_goals
    table_df.loc[table_df["Club"] == away_team, "GA"] += home_goals
    
table_df
    

Club  Won  Drawn  Lost  GF  GA  GD  Points
1        Liverpool   16      3     0  43   7   0      51
2        Tottenham   15      0     4  42  18   0      45
3         Man City   14      2     3  51  15   0      44
4          Chelsea   12      4     3  37  16   0      40
5          Arsenal   11      5     3  41  25   0      38
6       Man United    9      5     5  37  31   0      32
7        Leicester    8      4     7  24  22   0      28
8          Everton    7      6     6  31  29   0      27
9          Watford    8      3     8  26  27   0      27
10        West Ham    8      3     8  27  28   0      27
11     Bournemouth    8      2     9  27  33   0      26
12          Wolves    7      5     7  20  22   0      26
13        Brighton    6      4     9  21  27   0      22
14  Crystal Palace    5      4    10  17  25   0      19
15       Newcastle    4      5    10  14  26   0      17
16         Cardiff    4      3    12  18  38   0      15
17     Southampton    3      6    10  20  35   0      15
18         Burnley    3      3    13  17  41   0      12
19          Fulham    2      5    12  17  43   0      11
20    Huddersfield    2      4    13  12  34   0      10

    One of the most crucial part of the league table is the league format explaining the ranks of the teams that have same points. While we are calculating the goal differences we must also take into account this situation. The FA , in the Premier League's website, states that:
    "If any clubs finish with the same number of points, their position in the Premier League table is determined by goal difference, then the number of goals scored... "
    So we will determine the ranking according to this sentence.
    

In [12]:
for _,row in table_df.iterrows():
    club = row["Club"]
    goals_for = row["GF"]
    goals_against = row["GA"]
    
    table_df.loc[table_df["Club"] == club, "GD"] = goals_for - goals_against
    table_df = table_df.sort_values(by=["Points", "GD", "GF"], ascending = [False, False, False])

table_df.index = position
table_df

Club  Won  Drawn  Lost  GF  GA  GD  Points
1        Liverpool   16      3     0  43   7  36      51
2        Tottenham   15      0     4  42  18  24      45
3         Man City   14      2     3  51  15  36      44
4          Chelsea   12      4     3  37  16  21      40
5          Arsenal   11      5     3  41  25  16      38
6       Man United    9      5     5  37  31   6      32
7        Leicester    8      4     7  24  22   2      28
8          Everton    7      6     6  31  29   2      27
9         West Ham    8      3     8  27  28  -1      27
10         Watford    8      3     8  26  27  -1      27
11          Wolves    7      5     7  20  22  -2      26
12     Bournemouth    8      2     9  27  33  -6      26
13        Brighton    6      4     9  21  27  -6      22
14  Crystal Palace    5      4    10  17  25  -8      19
15       Newcastle    4      5    10  14  26 -12      17
16     Southampton    3      6    10  20  35 -15      15
17         Cardiff    4      3    12  18  38 -20      15
18         Burnley    3      3    13  17  41 -24      12
19          Fulham    2      5    12  17  43 -26      11
20    Huddersfield    2      4    13  12  34 -22      10

## Calculating the shooting efficiency of teams (Home and Away matches separately)

- HSE = Home Team Shooting Efficiency
- ASE = Away Team Shooting Efficiency


    We should consider these tables different by each other. For example, the "Home Games" table is shaped according to the home teams. The "HS" -means "Home Team Shots"- column contains the average number of shots taken by the home team against their opponents in the first half of the league. And "AS" -means "Away Team Shots"- column contains the average number of shots taken by their opponent to their goal.
    
    The "Away Games" table should be read in the opposite way.
    
    For both the "Home Games" and "Away Games" tables, there will be two tables each containing home and away games for all teams ranked by "Home Team Shooting Efficiency" and "Away Team Shooting Efficiency". These statistics are calculated as the ratio of the team's number of goals to shot attempts.
    
    We cannot talk about marginal benefits when interpreting this statistic. For example, one team may have a better goalkeeper or a team's defensive style may be much stronger than the others. In fact, this is a primitive statistics. It contains many parameters. It can show some important things but cannot fully explain them.

### Home Games

In [13]:
home_shooting_stats = df.groupby("HomeTeam")["HS", "AS", "HST", "AST", "FTHG", "FTAG"].mean()
home_shooting_stats

HS         AS       HST       AST      FTHG      FTAG
HomeTeam                                                                    
Arsenal         13.100000  10.900000  4.800000  4.200000  2.000000  0.900000
Bournemouth     11.700000  13.400000  4.000000  4.500000  1.600000  1.400000
Brighton         9.000000  13.333333  3.111111  4.555556  1.555556  1.222222
Burnley         10.111111  17.111111  3.555556  5.777778  1.111111  2.222222
Cardiff         13.900000  12.600000  3.400000  5.500000  1.300000  2.000000
Chelsea         16.800000   9.300000  6.300000  3.300000  1.900000  0.800000
Crystal Palace  16.666667  10.111111  4.222222  3.000000  0.555556  0.888889
Everton         15.000000   9.800000  4.400000  3.700000  1.800000  1.500000
Fulham          16.555556  11.777778  5.666667  5.555556  1.222222  2.111111
Huddersfield    11.100000  10.900000  3.100000  4.100000  0.400000  1.400000
Leicester       12.777778  10.000000  4.444444  3.000000  1.333333  1.000000
Liverpool       18.444444   6.111111  6.444444  2.000000  2.444444  0.222222
Man City        21.900000   6.200000  7.800000  2.000000  3.500000  0.900000
Man United      15.111111  11.333333  7.777778  4.777778  1.888889  1.333333
Newcastle       12.300000  11.600000  2.900000  3.400000  0.700000  1.500000
Southampton     14.777778  12.777778  5.000000  4.555556  1.111111  1.555556
Tottenham       14.375000  12.125000  6.125000  4.500000  2.125000  0.750000
Watford         12.200000  10.300000  4.400000  4.600000  1.500000  1.700000
West Ham        13.000000  10.600000  5.100000  4.600000  1.400000  1.600000
Wolves          13.000000  12.600000  4.000000  4.800000  1.200000  1.300000

In [14]:
home_shooting_stats["HSE"] = 0
home_shooting_stats["ASE"] = 0

for _,row in home_shooting_stats.iterrows():
    hs = row["HS"]
    as_ = row["AS"]
    hst = row["HST"]
    ast = row["AST"]
    fthg = row["FTHG"]
    ftag = row["FTAG"]
    hse = row["HSE"]
    ase = row["ASE"]
    
    home_shooting_stats.loc[row.name, "HSE"] = fthg/hs
    home_shooting_stats.loc[row.name, "ASE"] = ftag/as_
    
home_shooting_stats.sort_values(by="HSE", ascending=False)

HS         AS       HST       AST      FTHG      FTAG  \
HomeTeam                                                                       
Brighton         9.000000  13.333333  3.111111  4.555556  1.555556  1.222222   
Man City        21.900000   6.200000  7.800000  2.000000  3.500000  0.900000   
Arsenal         13.100000  10.900000  4.800000  4.200000  2.000000  0.900000   
Tottenham       14.375000  12.125000  6.125000  4.500000  2.125000  0.750000   
Bournemouth     11.700000  13.400000  4.000000  4.500000  1.600000  1.400000   
Liverpool       18.444444   6.111111  6.444444  2.000000  2.444444  0.222222   
Man United      15.111111  11.333333  7.777778  4.777778  1.888889  1.333333   
Watford         12.200000  10.300000  4.400000  4.600000  1.500000  1.700000   
Everton         15.000000   9.800000  4.400000  3.700000  1.800000  1.500000   
Chelsea         16.800000   9.300000  6.300000  3.300000  1.900000  0.800000   
Burnley         10.111111  17.111111  3.555556  5.777778  1.111111  2.222222   
West Ham        13.000000  10.600000  5.100000  4.600000  1.400000  1.600000   
Leicester       12.777778  10.000000  4.444444  3.000000  1.333333  1.000000   
Cardiff         13.900000  12.600000  3.400000  5.500000  1.300000  2.000000   
Wolves          13.000000  12.600000  4.000000  4.800000  1.200000  1.300000   
Southampton     14.777778  12.777778  5.000000  4.555556  1.111111  1.555556   
Fulham          16.555556  11.777778  5.666667  5.555556  1.222222  2.111111   
Newcastle       12.300000  11.600000  2.900000  3.400000  0.700000  1.500000   
Huddersfield    11.100000  10.900000  3.100000  4.100000  0.400000  1.400000   
Crystal Palace  16.666667  10.111111  4.222222  3.000000  0.555556  0.888889   

                     HSE       ASE  
HomeTeam                            
Brighton        0.172840  0.091667  
Man City        0.159817  0.145161  
Arsenal         0.152672  0.082569  
Tottenham       0.147826  0.061856  
Bournemouth     0.136752  0.104478  
Liverpool       0.132530  0.036364  
Man United      0.125000  0.117647  
Watford         0.122951  0.165049  
Everton         0.120000  0.153061  
Chelsea         0.113095  0.086022  
Burnley         0.109890  0.129870  
West Ham        0.107692  0.150943  
Leicester       0.104348  0.100000  
Cardiff         0.093525  0.158730  
Wolves          0.092308  0.103175  
Southampton     0.075188  0.121739  
Fulham          0.073826  0.179245  
Newcastle       0.056911  0.129310  
Huddersfield    0.036036  0.128440  
Crystal Palace  0.033333  0.087912

    In this table, Brighton looks the most interesting team. They shoot less than the other 19 teams while they have the highest target rating. Man City's statistics seem more sustainable than Brighton's. Both the number of shot attempts and scoring rates are quite high. They could benefit from this in the long run more than Brighton's unexceptional rating.

In [15]:
home_shooting_stats.sort_values(by="ASE", ascending=True)

HS         AS       HST       AST      FTHG      FTAG  \
HomeTeam                                                                       
Liverpool       18.444444   6.111111  6.444444  2.000000  2.444444  0.222222   
Tottenham       14.375000  12.125000  6.125000  4.500000  2.125000  0.750000   
Arsenal         13.100000  10.900000  4.800000  4.200000  2.000000  0.900000   
Chelsea         16.800000   9.300000  6.300000  3.300000  1.900000  0.800000   
Crystal Palace  16.666667  10.111111  4.222222  3.000000  0.555556  0.888889   
Brighton         9.000000  13.333333  3.111111  4.555556  1.555556  1.222222   
Leicester       12.777778  10.000000  4.444444  3.000000  1.333333  1.000000   
Wolves          13.000000  12.600000  4.000000  4.800000  1.200000  1.300000   
Bournemouth     11.700000  13.400000  4.000000  4.500000  1.600000  1.400000   
Man United      15.111111  11.333333  7.777778  4.777778  1.888889  1.333333   
Southampton     14.777778  12.777778  5.000000  4.555556  1.111111  1.555556   
Huddersfield    11.100000  10.900000  3.100000  4.100000  0.400000  1.400000   
Newcastle       12.300000  11.600000  2.900000  3.400000  0.700000  1.500000   
Burnley         10.111111  17.111111  3.555556  5.777778  1.111111  2.222222   
Man City        21.900000   6.200000  7.800000  2.000000  3.500000  0.900000   
West Ham        13.000000  10.600000  5.100000  4.600000  1.400000  1.600000   
Everton         15.000000   9.800000  4.400000  3.700000  1.800000  1.500000   
Cardiff         13.900000  12.600000  3.400000  5.500000  1.300000  2.000000   
Watford         12.200000  10.300000  4.400000  4.600000  1.500000  1.700000   
Fulham          16.555556  11.777778  5.666667  5.555556  1.222222  2.111111   

                     HSE       ASE  
HomeTeam                            
Liverpool       0.132530  0.036364  
Tottenham       0.147826  0.061856  
Arsenal         0.152672  0.082569  
Chelsea         0.113095  0.086022  
Crystal Palace  0.033333  0.087912  
Brighton        0.172840  0.091667  
Leicester       0.104348  0.100000  
Wolves          0.092308  0.103175  
Bournemouth     0.136752  0.104478  
Man United      0.125000  0.117647  
Southampton     0.075188  0.121739  
Huddersfield    0.036036  0.128440  
Newcastle       0.056911  0.129310  
Burnley         0.109890  0.129870  
Man City        0.159817  0.145161  
West Ham        0.107692  0.150943  
Everton         0.120000  0.153061  
Cardiff         0.093525  0.158730  
Watford         0.122951  0.165049  
Fulham          0.073826  0.179245

    It can be said that Liverpool is by far the first in terms of not giving a clear goal opportunity or having a high goalkeeper quality. 
    It's clear that Man City suffer from this statistic despite being third in the league table. They should work on the goals they conceded at home. It's a very high ratio.

### Away Games

In [16]:
away_shooting_stats = df.groupby("AwayTeam")["HS", "AS", "HST", "AST", "FTHG", "FTAG"].mean()
away_shooting_stats

HS         AS       HST       AST      FTHG      FTAG
AwayTeam                                                                    
Arsenal         13.777778  12.111111  5.000000  4.888889  1.777778  2.333333
Bournemouth     13.888889  11.777778  4.555556  3.888889  2.111111  1.222222
Brighton        17.900000   8.600000  5.000000  2.700000  1.600000  0.700000
Burnley         21.400000   7.000000  7.000000  1.800000  2.100000  0.700000
Cardiff         17.111111   8.333333  6.444444  3.000000  2.000000  0.555556
Chelsea          8.888889  15.555556  2.888889  4.333333  0.888889  2.000000
Crystal Palace  14.200000   9.400000  4.700000  3.500000  1.700000  1.200000
Everton         12.666667  10.444444  4.666667  4.444444  1.555556  1.444444
Fulham          18.400000   8.200000  6.500000  2.900000  2.400000  0.600000
Huddersfield    14.111111  12.000000  5.444444  3.555556  2.222222  0.888889
Leicester       12.900000  10.900000  4.100000  4.200000  1.300000  1.200000
Liverpool       10.000000  13.300000  3.300000  6.100000  0.500000  2.100000
Man City         7.888889  13.777778  3.111111  5.888889  0.666667  1.777778
Man United      15.400000  11.300000  5.100000  5.100000  1.900000  2.000000
Newcastle       17.333333   8.777778  5.222222  3.777778  1.222222  0.777778
Southampton     13.900000  14.000000  5.500000  4.700000  2.100000  1.000000
Tottenham       13.181818  11.545455  4.272727  4.818182  1.090909  2.272727
Watford         11.444444  11.555556  3.333333  3.777778  1.111111  1.222222
West Ham        15.444444  10.666667  5.888889  3.888889  1.333333  1.444444
Wolves          10.777778  13.666667  3.888889  5.111111  1.000000  0.888889

In [17]:
away_shooting_stats["HSE"] = 0
away_shooting_stats["ASE"] = 0

for _,row in away_shooting_stats.iterrows():
    hs = row["HS"]
    as_ = row["AS"]
    hst = row["HST"]
    ast = row["HST"]
    fthg = row["FTHG"]
    ftag = row["FTAG"]
    hse = row["HSE"]
    ase = row["ASE"]

    
    away_shooting_stats.loc[row.name, "HSE"] = fthg/hs
    away_shooting_stats.loc[row.name, "ASE"] = ftag/as_
    

away_shooting_stats.sort_values(by="ASE", ascending=False)

HS         AS       HST       AST      FTHG      FTAG  \
AwayTeam                                                                       
Tottenham       13.181818  11.545455  4.272727  4.818182  1.090909  2.272727   
Arsenal         13.777778  12.111111  5.000000  4.888889  1.777778  2.333333   
Man United      15.400000  11.300000  5.100000  5.100000  1.900000  2.000000   
Liverpool       10.000000  13.300000  3.300000  6.100000  0.500000  2.100000   
Everton         12.666667  10.444444  4.666667  4.444444  1.555556  1.444444   
West Ham        15.444444  10.666667  5.888889  3.888889  1.333333  1.444444   
Man City         7.888889  13.777778  3.111111  5.888889  0.666667  1.777778   
Chelsea          8.888889  15.555556  2.888889  4.333333  0.888889  2.000000   
Crystal Palace  14.200000   9.400000  4.700000  3.500000  1.700000  1.200000   
Leicester       12.900000  10.900000  4.100000  4.200000  1.300000  1.200000   
Watford         11.444444  11.555556  3.333333  3.777778  1.111111  1.222222   
Bournemouth     13.888889  11.777778  4.555556  3.888889  2.111111  1.222222   
Burnley         21.400000   7.000000  7.000000  1.800000  2.100000  0.700000   
Newcastle       17.333333   8.777778  5.222222  3.777778  1.222222  0.777778   
Brighton        17.900000   8.600000  5.000000  2.700000  1.600000  0.700000   
Huddersfield    14.111111  12.000000  5.444444  3.555556  2.222222  0.888889   
Fulham          18.400000   8.200000  6.500000  2.900000  2.400000  0.600000   
Southampton     13.900000  14.000000  5.500000  4.700000  2.100000  1.000000   
Cardiff         17.111111   8.333333  6.444444  3.000000  2.000000  0.555556   
Wolves          10.777778  13.666667  3.888889  5.111111  1.000000  0.888889   

                     HSE       ASE  
AwayTeam                            
Tottenham       0.082759  0.196850  
Arsenal         0.129032  0.192661  
Man United      0.123377  0.176991  
Liverpool       0.050000  0.157895  
Everton         0.122807  0.138298  
West Ham        0.086331  0.135417  
Man City        0.084507  0.129032  
Chelsea         0.100000  0.128571  
Crystal Palace  0.119718  0.127660  
Leicester       0.100775  0.110092  
Watford         0.097087  0.105769  
Bournemouth     0.152000  0.103774  
Burnley         0.098131  0.100000  
Newcastle       0.070513  0.088608  
Brighton        0.089385  0.081395  
Huddersfield    0.157480  0.074074  
Fulham          0.130435  0.073171  
Southampton     0.151079  0.071429  
Cardiff         0.116883  0.066667  
Wolves          0.092784  0.065041

    Statistics show why Tottenham is in second place in the league.

In [18]:
away_shooting_stats.sort_values(by="HSE", ascending=True)

HS         AS       HST       AST      FTHG      FTAG  \
AwayTeam                                                                       
Liverpool       10.000000  13.300000  3.300000  6.100000  0.500000  2.100000   
Newcastle       17.333333   8.777778  5.222222  3.777778  1.222222  0.777778   
Tottenham       13.181818  11.545455  4.272727  4.818182  1.090909  2.272727   
Man City         7.888889  13.777778  3.111111  5.888889  0.666667  1.777778   
West Ham        15.444444  10.666667  5.888889  3.888889  1.333333  1.444444   
Brighton        17.900000   8.600000  5.000000  2.700000  1.600000  0.700000   
Wolves          10.777778  13.666667  3.888889  5.111111  1.000000  0.888889   
Watford         11.444444  11.555556  3.333333  3.777778  1.111111  1.222222   
Burnley         21.400000   7.000000  7.000000  1.800000  2.100000  0.700000   
Chelsea          8.888889  15.555556  2.888889  4.333333  0.888889  2.000000   
Leicester       12.900000  10.900000  4.100000  4.200000  1.300000  1.200000   
Cardiff         17.111111   8.333333  6.444444  3.000000  2.000000  0.555556   
Crystal Palace  14.200000   9.400000  4.700000  3.500000  1.700000  1.200000   
Everton         12.666667  10.444444  4.666667  4.444444  1.555556  1.444444   
Man United      15.400000  11.300000  5.100000  5.100000  1.900000  2.000000   
Arsenal         13.777778  12.111111  5.000000  4.888889  1.777778  2.333333   
Fulham          18.400000   8.200000  6.500000  2.900000  2.400000  0.600000   
Southampton     13.900000  14.000000  5.500000  4.700000  2.100000  1.000000   
Bournemouth     13.888889  11.777778  4.555556  3.888889  2.111111  1.222222   
Huddersfield    14.111111  12.000000  5.444444  3.555556  2.222222  0.888889   

                     HSE       ASE  
AwayTeam                            
Liverpool       0.050000  0.157895  
Newcastle       0.070513  0.088608  
Tottenham       0.082759  0.196850  
Man City        0.084507  0.129032  
West Ham        0.086331  0.135417  
Brighton        0.089385  0.081395  
Wolves          0.092784  0.065041  
Watford         0.097087  0.105769  
Burnley         0.098131  0.100000  
Chelsea         0.100000  0.128571  
Leicester       0.100775  0.110092  
Cardiff         0.116883  0.066667  
Crystal Palace  0.119718  0.127660  
Everton         0.122807  0.138298  
Man United      0.123377  0.176991  
Arsenal         0.129032  0.192661  
Fulham          0.130435  0.073171  
Southampton     0.151079  0.071429  
Bournemouth     0.152000  0.103774  
Huddersfield    0.157480  0.074074

    Looking at both the home and away statistics of the opponents, it can be seen that Liverpool is the team with the strongest defense in this league.
    Man City's defensive statistics is better at away games than the home games.

In [19]:
df[["HS", "AS", "HST", "AST", "FTHG", "FTAG"]].mean()

HS      14.084211
AS      11.115789
HST      4.805263
AST      4.115789
FTHG     1.531579
FTAG     1.321053
dtype: float64

## Deciding the most "brutal" team in the Premier League's 2018-2019 season's first half

In [20]:
home_fouls_cards = df.groupby("HomeTeam")["HF", "HY", "HR"].mean()
home_fouls_cards.sort_values(by="HF", ascending = False)

HF        HY        HR
HomeTeam                                     
Brighton        12.666667  2.333333  0.111111
Wolves          12.500000  1.600000  0.000000
Arsenal         11.900000  2.200000  0.000000
Everton         11.600000  0.800000  0.000000
Man United      11.555556  1.666667  0.000000
Fulham          11.444444  1.666667  0.111111
Newcastle       11.400000  1.400000  0.100000
Southampton     11.333333  2.333333  0.111111
Bournemouth     10.400000  2.200000  0.100000
Tottenham       10.375000  0.750000  0.000000
Leicester       10.333333  2.111111  0.222222
Cardiff         10.300000  1.900000  0.000000
Burnley         10.222222  1.777778  0.000000
Huddersfield    10.100000  1.000000  0.200000
Watford         10.000000  1.500000  0.100000
West Ham         9.700000  1.900000  0.000000
Crystal Palace   9.555556  1.222222  0.111111
Liverpool        9.000000  0.777778  0.000000
Chelsea          8.900000  1.200000  0.000000
Man City         8.500000  0.500000  0.000000

In [21]:
away_fouls_cards = df.groupby("AwayTeam")["AF", "AY", "AR"].mean()
away_fouls_cards.sort_values(by="AF", ascending = False)

AF        AY        AR
AwayTeam                                     
Brighton        14.900000  1.500000  0.200000
Man United      13.000000  2.200000  0.200000
Watford         12.444444  1.888889  0.111111
Newcastle       12.333333  1.444444  0.111111
Huddersfield    12.222222  2.111111  0.000000
Wolves          12.000000  1.888889  0.000000
Crystal Palace  11.900000  2.300000  0.000000
Fulham          11.600000  1.600000  0.100000
Tottenham       11.000000  1.727273  0.090909
Everton         10.888889  2.000000  0.222222
Arsenal         10.555556  1.555556  0.000000
Southampton     10.200000  2.000000  0.000000
Burnley          9.700000  2.300000  0.000000
Liverpool        9.300000  0.900000  0.100000
Cardiff          9.222222  1.222222  0.111111
Leicester        9.200000  1.200000  0.200000
Man City         8.777778  1.444444  0.111111
Chelsea          8.777778  1.555556  0.000000
West Ham         8.666667  2.000000  0.111111
Bournemouth      7.111111  1.111111  0.000000

    It is clearly seen that Brighton played the toughest matches. It seems that they have a mentality that aims not letting the opponents play football.
    Bournemouth's attitude towards home and away games looks different. While they prefer a softer game in their away games, their harshness in home games approaches the league's fouling standards.

In [22]:
df[["HF", "AF", "HY", "AY", "HR", "AR"]].mean()

HF    10.584211
AF    10.715789
HY     1.542105
AY     1.700000
HR     0.057895
AR     0.084211
dtype: float64

## Examining the strictness of the referees

In [23]:
ref_stats = df.groupby("Referee")[["HF", "AF", "HY", "AY", "HR", "AR"]].mean()
ref_stats["Matches"] = 0
matches = df.value_counts("Referee")


ref_stats["Matches"] = df["Referee"].value_counts()
    
ref_stats.sort_values(by="Matches", ascending = False)

HF         AF        HY        AY        HR        AR  \
Referee                                                                    
A Taylor    11.058824   9.941176  1.588235  1.411765  0.000000  0.000000   
M Atkinson  11.187500  11.750000  1.250000  1.812500  0.000000  0.062500   
M Oliver    10.562500  10.937500  1.125000  1.562500  0.187500  0.062500   
A Marriner   9.200000   8.266667  1.200000  1.733333  0.133333  0.000000   
M Dean       9.933333  12.000000  1.933333  2.000000  0.133333  0.333333   
C Pawson    10.857143  10.714286  2.142857  1.642857  0.000000  0.214286   
J Moss      11.692308  11.000000  1.692308  1.461538  0.153846  0.153846   
K Friend    10.923077  12.307692  1.923077  2.153846  0.076923  0.000000   
C Kavanagh   9.666667   9.583333  1.916667  1.250000  0.000000  0.083333   
P Tierney   10.500000  11.833333  1.833333  2.000000  0.000000  0.000000   
L Mason     10.800000  12.800000  1.300000  2.400000  0.000000  0.000000   
S Attwell   10.500000   7.900000  1.900000  1.300000  0.000000  0.000000   
L Probert   10.800000  11.700000  1.000000  1.800000  0.100000  0.200000   
G Scott      9.500000   9.166667  0.666667  1.500000  0.000000  0.166667   
D Coote     10.500000  12.000000  1.000000  0.750000  0.000000  0.000000   
S Hooper    13.000000   8.333333  1.333333  1.333333  0.000000  0.000000   
R East      11.333333  11.000000  1.666667  3.000000  0.000000  0.000000   
A Madley     7.000000   5.000000  0.000000  0.000000  0.000000  0.000000   

            Matches  
Referee              
A Taylor         17  
M Atkinson       16  
M Oliver         16  
A Marriner       15  
M Dean           15  
C Pawson         14  
J Moss           13  
K Friend         13  
C Kavanagh       12  
P Tierney        12  
L Mason          10  
S Attwell        10  
L Probert        10  
G Scott           6  
D Coote           4  
S Hooper          3  
R East            3  
A Madley          1

    "Matches" column states the number of managed matches for each referee.
    Since it will not be very meaningful and it will be difficult to evaluate it separately, we will evaluate it by adding the numbers of the home and away teams.

- F = Fouls
- YC = Yellow Cards
- RC = Red Cards
- CPF = Cards per Foul

In [24]:
ref_stats["F"] = 0
ref_stats["YC"] = 0
ref_stats["RC"] = 0
 
for _,row in ref_stats.iterrows():
    hf = row["HF"]
    af = row["AF"]
    hy = row["HY"]
    ay = row["AY"]
    hr = row["HR"]
    ar = row["AR"]
    

    
    ref_stats.loc[row.name, "F"] = hf + af
    ref_stats.loc[row.name, "YC"] = hy + ay
    ref_stats.loc[row.name, "RC"] = hr + ar
    
ref_stats = ref_stats.drop(columns=["HF", "AF", "HY", "AY", "HR", "AR"])
ref_stats.sort_values(by="Matches", ascending = False)

Matches          F        YC        RC
Referee                                           
A Taylor         17  21.000000  3.000000  0.000000
M Atkinson       16  22.937500  3.062500  0.062500
M Oliver         16  21.500000  2.687500  0.250000
A Marriner       15  17.466667  2.933333  0.133333
M Dean           15  21.933333  3.933333  0.466667
C Pawson         14  21.571429  3.785714  0.214286
J Moss           13  22.692308  3.153846  0.307692
K Friend         13  23.230769  4.076923  0.076923
C Kavanagh       12  19.250000  3.166667  0.083333
P Tierney        12  22.333333  3.833333  0.000000
L Mason          10  23.600000  3.700000  0.000000
S Attwell        10  18.400000  3.200000  0.000000
L Probert        10  22.500000  2.800000  0.300000
G Scott           6  18.666667  2.166667  0.166667
D Coote           4  22.500000  1.750000  0.000000
S Hooper          3  21.333333  2.666667  0.000000
R East            3  22.333333  4.666667  0.000000
A Madley          1  12.000000  0.000000  0.000000

In [25]:
for _,row in ref_stats.iterrows():
    f = row["F"]
    yc = row["YC"]
    rc = row["RC"]
    
    ref_stats.loc[row.name, "CPF"] = (yc + 2*rc) / f
    
ref_stats.sort_values(by = "CPF", ascending = False)

Matches          F        YC        RC       CPF
Referee                                                     
M Dean           15  21.933333  3.933333  0.466667  0.221884
R East            3  22.333333  4.666667  0.000000  0.208955
C Pawson         14  21.571429  3.785714  0.214286  0.195364
A Marriner       15  17.466667  2.933333  0.133333  0.183206
K Friend         13  23.230769  4.076923  0.076923  0.182119
S Attwell        10  18.400000  3.200000  0.000000  0.173913
C Kavanagh       12  19.250000  3.166667  0.083333  0.173160
P Tierney        12  22.333333  3.833333  0.000000  0.171642
J Moss           13  22.692308  3.153846  0.307692  0.166102
L Mason          10  23.600000  3.700000  0.000000  0.156780
L Probert        10  22.500000  2.800000  0.300000  0.151111
M Oliver         16  21.500000  2.687500  0.250000  0.148256
A Taylor         17  21.000000  3.000000  0.000000  0.142857
M Atkinson       16  22.937500  3.062500  0.062500  0.138965
G Scott           6  18.666667  2.166667  0.166667  0.133929
S Hooper          3  21.333333  2.666667  0.000000  0.125000
D Coote           4  22.500000  1.750000  0.000000  0.077778
A Madley          1  12.000000  0.000000  0.000000  0.000000

    It looks Mike Dean is at the top of the table as the most strict referee in the league with showing 0.221 cards per foul. That is nearly equal to showing a card per 4.5 fouls. Besides Andrew Madley is at the bottom. But he had managed only one match for the first half of the season. So it can be evaluated as an outlier. There could be many reasons for this statistic. Let's have a look at the match he managed to interpret more precisely.

In [26]:
df[df["Referee"] == "A Madley"]

Date HomeTeam AwayTeam  FTHG  FTAG FTR  HTHG  HTAG HTR   Referee  \
165  15/12/18  Watford  Cardiff     3     2   H     1     0   H  A Madley   

     ...  HST  AST  HF  AF  HC  AC  HY  AY  HR  AR  
165  ...    8    3   7   5   5   0   0   0   0   0  

[1 rows x 22 columns]

    Only match managed by Andrew Madley is Watford against Cardiff at home. If we take a look at the fouls and cards statistics of these teams while Watford committed 10.0 fouls on average when playing at home, Cardiff committed 9.2 fouls in their away games. The leagues average fouls numbers are 10.584211 in home games and 10.715789 in away games. In other words, both teams commit fouls below the league average anyway. This reason could be effected the number of cards Madley showed in that match. But still, there's also the possibility that Madley is a soft referee. However, he needs to manage more matches to consider him on this list.

In [27]:
ref_stats.sort_values(by = "F", ascending = False)

Matches          F        YC        RC       CPF
Referee                                                     
L Mason          10  23.600000  3.700000  0.000000  0.156780
K Friend         13  23.230769  4.076923  0.076923  0.182119
M Atkinson       16  22.937500  3.062500  0.062500  0.138965
J Moss           13  22.692308  3.153846  0.307692  0.166102
D Coote           4  22.500000  1.750000  0.000000  0.077778
L Probert        10  22.500000  2.800000  0.300000  0.151111
R East            3  22.333333  4.666667  0.000000  0.208955
P Tierney        12  22.333333  3.833333  0.000000  0.171642
M Dean           15  21.933333  3.933333  0.466667  0.221884
C Pawson         14  21.571429  3.785714  0.214286  0.195364
M Oliver         16  21.500000  2.687500  0.250000  0.148256
S Hooper          3  21.333333  2.666667  0.000000  0.125000
A Taylor         17  21.000000  3.000000  0.000000  0.142857
C Kavanagh       12  19.250000  3.166667  0.083333  0.173160
G Scott           6  18.666667  2.166667  0.166667  0.133929
S Attwell        10  18.400000  3.200000  0.000000  0.173913
A Marriner       15  17.466667  2.933333  0.133333  0.183206
A Madley          1  12.000000  0.000000  0.000000  0.000000

In [28]:
ref_standards = ref_stats[["F", "YC", "RC", "CPF"]].mean()
ref_standards

F      20.847148
YC      3.032397
RC      0.114522
CPF     0.152834
dtype: float64

    The average values are above. It can be decided which referees are strict or soft relatively to these numbers. 
    For example, if a referee who whistles a lot of fouls is officiating the match, we can predict that the match will be less entertaining to watch. In contrast, if a referee whistles for fewer fouls on average, the tempo of the match can be said to be higher.
    